resolver/resolver.go

```diff
@@ -107,16 +129,20 @@ type Address struct {
 // parameter, and the resolver will return only IP addresses of the type
 // specified by network. The network must be one of "ip", "ip4" or "ip6".
 // Note that because net.Resolver does not expose the record TTL values, this
-// resolver uses the fixed TTL provided in the ttl parameter.
+// resolver uses the fixed TTL provided in the ttl parameter. The specified
+// address family affinity value can be used to prefer using either IPv4 or
+// IPv6 addresses only, in cases where there are both A and AAAA records.
 func NewDNSResolver(
 	resolver *net.Resolver,
 	network string,
 	ttl time.Duration,
+	affinity AddressFamilyAffinity,
```

Comment:
I just had a thought: the affinity is only useful when combined with "ip" network, right?

Maybe we should _remove_ the `network` argument and expand the affinity settings to include two new ones: `RequireIPv4` and `RequireIPv6`. Under the hood, the latter two would result in "ip4" or "ip6" when actually making the `LookupNetIP` call.

Reply:
Done, now `policy` replaces `network` and `affinity`.